### PR TITLE
Developer 1775 downloads ac test - B

### PR DESCRIPTION
**Pull request by RubyGemTSL**
minor update to env.rb

updated driver names to match jenkins

updated raven due to issues testing on firefox.

### CODE REVIEW

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by paulrobinson** [comment]
Let me know when this is ready for review.

**Comment by RubyGemTSL** [comment]
@paulrobinson something is not quite right with the acceptance test job. It seems like its using master rather than my branch :-/ as the download tests are not being ran :-/

**Comment by RubyGemTSL** [comment]
@paulrobinson I think I know whats up with it. Will ping you when its ready to review

**Comment by RubyGemTSL** [comment]
@paulrobinson - Finally I may be finished with this ticket! (pending review)

**Comment by paulrobinson** [comment]
Have you ensured that they don't run in Production?

**Comment by RubyGemTSL** [comment]
@paulrobinson these would only be ran in production if the acceptance tests for production had the smoke tag removed.

**Comment by paulrobinson** [comment]
You seem to use 'Customer' a lot. What's your definition of Customer?

**Comment by RubyGemTSL** [comment]
@paulrobinson  Customer = site visitor

File: _cucumber/features/downloads/cdk_downloads.feature [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  @customer
+  @logout
+  @javascript
+  Scenario: Newly registered RHD navigates to CDK Download page and clicks on download, upgrades account, who accepts Redhat T&C's should initiate CDK download.
+    Given I register a new account
+    When I am on the Download Overview page for cdk
+    And I click to download the featured download of "Red Hat Container Development Kit (CDK)"
+    When I upgrade my account and accept the terms and conditions
+    Then I should see the cdk get started page with a confirmation message Thank you for downloading Red Hat Container Development Kit (CDK)
```
Comment (paulrobinson) [inline _cucumber/features/downloads/cdk_downloads.feature]: Quote: Thank you for downloading Red Hat Container Development Kit (CDK)

**Comment by paulrobinson** [comment]
Probably best to replace 'customer' with 'user'. We want people outside of the engineering group to read these, so they need to be clear.

File: _cucumber/features/downloads/cdk_downloads.feature [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    And the download should initiate
+
+  @customer
+  @logout
+  @javascript
+  Scenario: Newly registered RHD navigates to CDK Download page and clicks on download, upgrades account, but does NOT accept Redhat T&C's should not initiate CDK download.
```
Comment (paulrobinson) [inline _cucumber/features/downloads/cdk_downloads.feature]: 'Newly registered RHD' -> 'Newly registered User'

There are similar occurrences of this elsewhere.

File: _cucumber/features/downloads/cdk_downloads.feature [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    Then I should see the cdk download overview page
+    And the download should not initiate
+
+  @customer
+  @logout
+  @javascript
+  Scenario: Unauthorised customer who has navigated directly to CDK Download overview page clicks on download, registers, upgrades account, and accepts Redhat T&C's should initiate CDK download.
+    Given I am on the Download Overview page for cdk
+    When I click to download the featured download of "Red Hat Container Development Kit (CDK)"
+    Then I should be redirected to the Developers.redhat Login Page
```
Comment (paulrobinson) [inline _cucumber/features/downloads/cdk_downloads.feature]: 'Developers.redhat' -> 'developers.redhat.com'

File: _cucumber/features/downloads/cdk_downloads.feature [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    When I click on the Create account link
+    And I complete the registration form
+    And I navigate to the verify account URL
```
Comment (paulrobinson) [inline _cucumber/features/downloads/cdk_downloads.feature]: Missing 'receive email' step

File: _cucumber/features/downloads/cdk_downloads.feature [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    And I upgrade my account and accept the terms and conditions
+    Then I should see the cdk get started page with a confirmation message Thank you for downloading Red Hat Container Development Kit (CDK)
```
Comment (paulrobinson) [inline _cucumber/features/downloads/cdk_downloads.feature]: Needs quotes: Thank you for downloading Red Hat Container Development Kit (CDK)

File: _cucumber/features/downloads/cdk_downloads.feature [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    And the download should initiate
+
+  @customer
+  @logout
+  @javascript
+  Scenario: Unauthorised customer who has navigated directly to CDK Download overview page clicks on download, registers, upgrades account, but does NOT accept Redhat T&C's should not initiate CDK download.
+    Given I am on the Download Overview page for cdk
+    When I click to download the featured download of "Red Hat Container Development Kit (CDK)"
+    Then I should be redirected to the Developers.redhat Login Page
```
Comment (paulrobinson) [inline _cucumber/features/downloads/cdk_downloads.feature]: 'Developers.redhat' -> 'RHD'. Same further up.

File: _cucumber/features/downloads/downloads.feature [hunk 7]
```diff
@@ -0,0 +1,20 @@
+Feature: Download Page - Unauthorised customer
+
+  In order to try out Red Hat software,
+  As generic site visitor,
+  I want to be able to see a list of available downloads.
+
+  @downloads
+  @javascript
+  Scenario: Unauthorised customer should see a list of available downloads from RedHat.
```
Comment (paulrobinson) [inline _cucumber/features/downloads/downloads.feature]: RedHat -> Red Hat

File: _cucumber/features/downloads/featured_downloads.feature [hunk 8]
```diff
@@ -0,0 +1,38 @@
+@downloads
+@product_download
+
+Feature: Download Page - An authorised customer downloads from available download manager resources.
```
Comment (paulrobinson) [inline _cucumber/features/downloads/featured_downloads.feature]: 'Download Page' -> 'Product Download Page'

File: _cucumber/features/downloads/featured_downloads.feature [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+  As a develpers.redhat.com site visitor,
+  I want to be able to register and download the Red Hat products.
+
+  @customer
+  @logout
+  @javascript
+  Scenario: Newly registered RHD navigates to Download tab and clicks on 'Download latest'
+    Given I register a new account
+    And I am on the Download Overview page for eap
```
Comment (paulrobinson) [inline _cucumber/features/downloads/featured_downloads.feature]: 'Download Overview' -> 'Product Download'

File: _cucumber/features/downloads/featured_downloads.feature [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    When I click to download the featured download of "Enterprise Application Server"
+    And I accept the terms and conditions
+    Then I should see the eap get started page with a confirmation message Thank you for downloading Enterprise Application Server
```
Comment (paulrobinson) [inline _cucumber/features/downloads/featured_downloads.feature]: Quotes: Thank you for downloading Enterprise Application Server

File: _cucumber/features/downloads/featured_downloads.feature [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    And the download should initiate
+
+  @customer
+  @logout
+  @javascript
+  Scenario: Newly registered RHD navigates to product Download page and clicks on download, enters their Company and Country but but does NOT accept Redhat T&C's should not initiate download.
```
Comment (paulrobinson) [inline _cucumber/features/downloads/featured_downloads.feature]: Needs updating, with changes to company and country.

File: _cucumber/lib/pages/company_and_country_page.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+require_relative 'base_page'
```
Comment (paulrobinson) [inline _cucumber/lib/pages/company_and_country_page.rb]: This file is not needed anymore.

File: _cucumber/lib/pages/download_overview_page.rb [hunk 13]
```diff
@@ -0,0 +1,28 @@
+require_relative 'base_page'
```
Comment (paulrobinson) [inline _cucumber/lib/pages/download_overview_page.rb]: Rename file to 'product_download_page.rb'

**Pull request merged**